guides/command_objects.md

```diff
@@ -0,0 +1,139 @@
+# Command Objects
+
+Implementation of [Command pattern](https://en.wikipedia.org/wiki/Command_pattern).
+
+## Conventions
+
+@TBD
+
+## Examples
+
+```ruby
+# app/commands/application_command.rb
+class ApplicationCommand
+  extend ActiveModel::Naming
+
+  class << self
+    def call(*args)
+      cmd = new(*args)
+      cmd.call
+    end
+  end
+
+  def call
+    _execute
+  end
+
+  def execute
+    raise NotImplementedError
+  end
+
+  def context
+    @_context ||= {}
+  end
+
+  def errors
+    @_errors ||= ActiveModel::Errors.new(self)
+  end
+
+  def fail!
+    @_success = false
+  end
+
+  def validate
```

Comment:
What kind of validations will be here?

Reply:
Anything that may or should be validated before command execution. But I haven't really played with use cases for validations yet. I've just outlined some methods like `validate`, `authorize` that might be useful.

Any thoughts are welcomed :)

Reply:
Well, i used separated validator objects. And it looked like this:
`
user = User.new(user_params)
if UserRegistrationVelidator.new(user).valid?
  UserRegistrationCommand.call(user)
else
 #errors processing
end
`
So, we will use command for business logic processing, and validators for data validations.
Also, im not sure, if we need to place authorize tools inside, but they use sometimes inside commands

Reply:
https://api.rubyonrails.org/classes/ActiveModel/Validator.html

Reply:
Seems that the purpose of `UserRegistrationValidator` is very close to [Form Objects](https://github.com/ergoserv/handbook/blob/master/guides/form_objects.md) we already used. The benefits of the `Form Object` approach is that it provides the object that can be nicely rendered on the view. 

As for the authorization tools. Sometimes commands may require to check the policy of the user, but probably you are right. We can remove this from basic command add some conventions or move authorization tools to a module, which could be included when needed.